src/AbsBackup.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\ProjectBackup;
+
+use GuzzleHttp\Client;
+use Keboola\StorageApi\Client as StorageApi;
+use Keboola\StorageApi\HandlerStack;
+use Keboola\StorageApi\Options\GetFileOptions;
+use Keboola\Temp\Temp;
+use MicrosoftAzure\Storage\Blob\BlobRestProxy;
+use Psr\Log\LoggerInterface;
+
+class AbsBackup extends Backup
+{
+    private BlobRestProxy $absClient;
+
+    private string $path;
+
+    public function __construct(
+        StorageApi $sapiClient,
+        BlobRestProxy $absClient,
+        string $path,
+        ?LoggerInterface $logger = null
+    ) {
+        $this->absClient = $absClient;
+        $this->path = $path;
+
+        parent::__construct($sapiClient, $logger);
+    }
+
+    protected function putToStorage(string $name, string $content): void
+    {
+        $this->absClient->createBlockBlob(
+            $this->path,
+            $name,
+            $content
+        );
+    }
+
+    public function backupTable(string $tableId): void
+    {
+        $table = $this->sapiClient->getTable($tableId);
+
+        if ($table['bucket']['stage'] === 'sys') {
```

Comment:
Tato podmienka by IMHO nemala byt priamo v `Abs` implementacii, plati vzdy rovnako aj dalsie + logovanie.

Abs/S3 interface by mali implementovat iba nevyhnutne metody, v ktorych sa lisia. Zvysok by mal byt v spolocnej triede (mensia nachylnost na chyby, lepsia udrzba .... ).

Vidim, ze tu mas pekne metodu `putToStorage`, no v kode pouzivas `$this->absClient->createBlockBlob`. 
Preco?

Reply:
1/ 👍 
2/ tak by to mělo být.. metoda `backupTable` se celkem liší na ABS/S3 tak proto má každy "adapter" napsaný svojí... ostatní metody jsou společné
3/ jop pravda

Reply:
ad 2/ `backupTable` jsem ještě trochu víc zjednodušil a získávání "fileInfo" jsem vyhodil zvlášť do metody